### PR TITLE
fix: Issue 11581

### PR DESCRIPTION
**Pull request by DominionSpy**
Fix for Issue #11581

### CODE REVIEW

File: Mage.Tests/src/test/java/org/mage/test/cards/copy/CopySpellTest.java [hunk 1]
```diff
@@ -778,6 +777,80 @@ public void copySpellWithKicker() {
...
+        // Copy target creature spell you control, except it isn’t legendary if the spell is legendary.
+        addCard(Zone.HAND, playerA, "Double Major");
+
+        activateManaAbility(1, PhaseStep.PRECOMBAT_MAIN, playerA, "<i>Psychic");
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Neverwinter Hydra");
+        setChoice(playerA, "X=2");
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Double Major", "Neverwinter Hydra");
+
+        setStopAt(1, PhaseStep.BEGIN_COMBAT);
```
Comment (xenohedron) [inline Mage.Tests/src/test/java/org/mage/test/cards/copy/CopySpellTest.java:801]: please include `setStrictChooseMode(true)` for all new tests to be more robust. Probably have to add a few `addTarget` and `setChoice` to clarify what's happening.

**Review by xenohedron — COMMENTED** [review]
Thanks for writing unit tests!

Using `CardUtil.getSourceCostsTag` definitely looks cleaner. I assume this was based off `EntersBattlefieldWithXCountersEffect`?

**Comment by DominionSpy** [comment]
> I assume this was based off `EntersBattlefieldWithXCountersEffect`?

Yes, that's right. I reasoned that there should be very little difference to that logic, save for the extra step of rolling dice or the Apocalypse Hydra check.

**Review by xenohedron — APPROVED** [review]
thanks for the contribution

**Pull request merged**